src/dosext/misc/emm.c

```diff
@@ -2062,7 +2062,12 @@ ems_fn(struct vm86_regs *state)
 	       ptr));
 
       for (i = 0; i < MAX_HANDLES; i++) {
-	if ((i == 0) || (handle_info[i].numpages > 0)) {
+	if ((i == 0) || (handle_info[i].active != 0)) {
+	  if (tot_handles == handle_total) {
+	    error("EMM: tot_handles == handle_total (0x%04X)"
```

Comment:
Can this trigger when i==0
and both tot_handles and handle_total == 0?
I prefer your prev patch.
Sanity checks on local data should
only print an error or assert, but not
alter the execution logic. Its not like
checking the user-provided parameters.

Reply:
I think that handle_total is always [initialised to one](https://github.com/dosemu2/dosemu2/blob/devel/src/dosext/misc/emm.c#L2255). If you prefer I can change it to allow the edge case (tomorrow).

Reply:
If handle_total is zero then actually we specified to the application that a zero size buffer would be enough to hold the result. It is an edge case but cancelling for handle_total zero is actually correct.

Reply:
I would prefer to avoid redundancy, and
make `tot_handles` for debug only. So that
it is only use to print error and nothing mode.
What do you think?

Reply:
I'd rather keep it as is.